### PR TITLE
Fix that reactive health endpoint is not handled correctly

### LINKED ISSUE #4396: WebOperationService incorrectly handling ReactiveHealthIndicator results
Spring Boot WebFlux applications with reactive health indicators are incorrectly returning HTTP 200 with the following response body when requesting application health: 
```json 
{
	"scanAvailable": true
}
```

After inspecting [WebOperationService#handleResult](https://github.com/line/armeria/blob/09ab81037f5a8a9e15e4d639474e7a925b0c40f5/spring/boot2-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/WebOperationService.java#L202) (lines 217-225 to be exact), the method is not accounting for the possibility that `result` is `instanceof Mono<Health>`. This results in response status code of 200 with `result` as the body instead of the value emitted by `result`.

JDK: 17.0.3-tem
Armeria version: 1.17.2/1.18.0
Spring Boot version: 2.7.2

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
If an autoconfigure module is in the classpath, [HealthEndpointWebExtension](https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/HealthEndpointWebExtension.java) is used for handling an actuator request.
Whereas, if a WebFlux module is also in the classpath, [ReactiveHealthEndpointWebExtension](https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/ReactiveHealthEndpointWebExtension.java) is used which returns `Mono`.
We should handle the `Mono` response.

Modifications:
- Handle `Mono` response in `WebOperationService`.

Result:
- Close #4396 
- You can now use Armeria Spring actuator with WebFlux correctly.

Co-authored-by: @DasAmpharos

### CODE REVIEW

File: spring/boot2-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/ReactiveHealthEndpointWebExtensionUtil.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        }
+
+        return handleResult(statusMapper, ctx, result, method);
+    }
+
+    private static HttpResponse handlePublisher(SimpleHttpCodeStatusMapper statusMapper,
+                                                ServiceRequestContext ctx, Publisher<?> result,
+                                                HttpMethod method) {
+        final CompletableFuture<HttpResponse> future = new CompletableFuture<>();
+        Mono.from(result).subscribe(result0 -> {
```
Comment (minwoox) [inline spring/boot2-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/ReactiveHealthEndpointWebExtensionUtil.java:54]: This logic is copied from [here](https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/web/reactive/AbstractWebFluxEndpointHandlerMapping.java#L373-L382)

Comment (ikhoon) [reply spring/boot2-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/ReactiveHealthEndpointWebExtensionUtil.java:54]: Not sure we really need backpressure here. But `.subscribe()` seems to break backpressure. 
Should we use `.map` and `StreamMessage.concat()` to connect the stream changes?
For example:
```java
Mono<HttpResponse> result = Mono.from(result)
                                .map(result0 ->  ... );
StreamMessage<HttpObject> concatenated = StreamMessage.concat(result);
return HttpResponse.of(concatenated);
```

File: spring/boot2-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/ReactiveHealthEndpointWebExtensionUtil.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+final class ReactiveHealthEndpointWebExtensionUtil {
+
+    static HttpResponse handleMaybeReactorResult(SimpleHttpCodeStatusMapper statusMapper,
+                                                 ServiceRequestContext ctx, Object result,
+                                                 HttpMethod method) throws Throwable {
+        if (result instanceof Flux) {
```
Comment (jrhee17) [inline spring/boot2-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/ReactiveHealthEndpointWebExtensionUtil.java:39]: nice catch 👍

**Review by jrhee17 — APPROVED** [review]
Looks good to me! Thanks @minwoox 🙇 👍 🙇

File: spring/boot2-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/ReactiveHealthEndpointWebExtensionUtil.java (hunk 2 above)
Comment (jrhee17) [reply spring/boot2-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/ReactiveHealthEndpointWebExtensionUtil.java:39]: Actually, I have a question.

If I understand correctly, `Flux` isn't actually returned anywhere yet.
https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/ReactiveHealthEndpointWebExtension.java

I'm wondering which is correct behavior when handling flux input:
1) Just return the first result and ignore the rest (like the default implementation of `Mono.from(Flux)`)
2) Collect all elements like done in the current PR

File: spring/boot2-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/ReactiveHealthEndpointWebExtensionUtil.java (hunk 1 above)
Comment (minwoox) [reply spring/boot2-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/ReactiveHealthEndpointWebExtensionUtil.java]: > Not sure we really need backpressure here.

We need that. 😄 Thanks for catching this. 👍 👍

File: spring/boot2-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/ReactiveHealthEndpointWebExtensionUtil.java (hunk 2 above)
Comment (minwoox) [reply spring/boot2-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/ReactiveHealthEndpointWebExtensionUtil.java:39]: > Flux isn't actually returned anywhere yet.

I couldn't find that either. 😅 

> I'm wondering which is correct behavior when handling flux input:

They just returned the collected list as a body with OK status so I followed it.
https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/web/reactive/AbstractWebFluxEndpointHandlerMapping.java#L385-L387
https://github.com/line/armeria/blob/master/spring/boot2-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/WebOperationService.java#L223-L225

File: spring/boot2-webflux-autoconfigure/src/test/java/com/linecorp/armeria/spring/web/reactive/ArmeriaSpringActuatorReactiveIndicatorTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    @LocalServerPort
+    int port;
+
+    private WebClient client;
+
+    @BeforeEach
+    void setUp() {
+        client = WebClient.of("http://127.0.0.1:" + port);
```
Comment (ikhoon) [inline spring/boot2-webflux-autoconfigure/src/test/java/com/linecorp/armeria/spring/web/reactive/ArmeriaSpringActuatorReactiveIndicatorTest.java:83]: nit: Use `BlockingWebClient`?

**Review by ikhoon — APPROVED** [review]
Nice work!

File: spring/boot2-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/ReactiveHealthEndpointWebExtensionUtil.java (hunk 2 above)
Comment (jrhee17) [reply spring/boot2-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/ReactiveHealthEndpointWebExtensionUtil.java:39]: Understood. Fair enough 👍

File: spring/boot2-webflux-autoconfigure/src/test/java/com/linecorp/armeria/spring/web/reactive/ArmeriaSpringActuatorReactiveIndicatorTest.java (hunk 3 above)
Comment (minwoox) [reply spring/boot2-webflux-autoconfigure/src/test/java/com/linecorp/armeria/spring/web/reactive/ArmeriaSpringActuatorReactiveIndicatorTest.java]: Ah, we have that. 😅

**Pull request merged**